### ISSUE TITLE
Async Function

### ISSUE THREAD

**Issue by hal-shu-sato** [hal-shu-sato/apm#312]
## Is your feature request related to a problem? Please describe.

Now, apm is mainly written using the sync function.
But I heard sync functions block the process.
Is it better to replace sync functions with async ones?
Please give your opinion.

**Comment by mitosagi** [comment]
I agree with you. I think it's better to start with the functions that are closest to the root, because halfway asynchronization can cause timing-dependent bugs that are difficult to debug.

**Comment by hal-shu-sato** [comment]
#320(Lib for File Operation)と実質同じ内容のため、こちらへ統合しました。

## 現況

起票時(2022-01)の「apm は主に同期関数で書かれている」という前提は、当時ほどは当たらなくなっています。ただし **`src/` に同期 fs 呼び出しが 37 件残っています**(テストを除く)。

多い順:

| ファイル | 件数 |
| --- | --- |
| `src/main/services/migration.ts` | 6 |
| `src/main/services/core.ts` | 5 |
| `src/shared/install.ts` | 4 |
| `src/main/services/scriptInstall.ts` | 3 |
| `src/main/services/packageUninstall.ts` | 3 |

## 補足

すべてが問題というわけではありません。`src/shared/unzip.ts` の `chmodSync` のように、モジュール評価時に一度だけ実行するものは同期のほうが素直です。一方 `install.ts` や `core.ts` のように**インストール処理の途中で繰り返し呼ばれるもの**は、非同期化の効果が見込めます。

一括で置き換えるより、触るファイルのついでに直していくのが現実的だと思われます。

---

調査と文面の作成に [Claude Code](https://claude.com/claude-code) を使用しています。
